recipes/libid3tag/all/conanfile.py

```diff
@@ -33,96 +27,32 @@ class LibId3TagConan(ConanFile):
-            self.tool_requires("gnu-config/cci.20210814")
-            if self.settings_build.os == "Windows":
-                self.win_bash = True
-                if not self.conf.get("tools.microsoft.bash:path", check_type=str):
-                    self.tool_requires("msys2/cci.latest")
-
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    def generate(self):
-        if is_msvc(self):
-            tc = CMakeToolchain(self)
-            tc.preprocessor_definitions["ID3TAG_EXPORT"] = "__declspec(dllexport)" if self.options.shared else ""
-            tc.generate()
-            deps = CMakeDeps(self)
-            deps.generate()
-        else:
-            venv = VirtualBuildEnv(self)
-            venv.generate()
-            tc = AutotoolsToolchain(self)
-            tc.generate()
-            deps = AutotoolsDeps(self)
-            deps.generate()
-
     def build(self):
-        if is_msvc(self):
-            # https://github.com/markjeee/libid3tag/blob/master/id3tag.h#L355-L358
-            replace_in_file(self, os.path.join(self.source_folder, "id3tag.h"),
-                            "extern char", "ID3TAG_EXPORT extern char")
-            cmake = CMake(self)
-            cmake.configure()
-            cmake.build()
-        else:
-            for gnu_config in [
-                self.conf.get("user.gnu-config:config_guess", check_type=str),
-                self.conf.get("user.gnu-config:config_sub", check_type=str),
-            ]:
-                if gnu_config:
-                    copy(self, os.path.basename(gnu_config), os.path.dirname(gnu_config), self.source_folder)
-            with chdir(self, self.source_folder):
-                autotools = Autotools(self)
-                autotools.configure()
-                autotools.make()
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
 
     def package(self):
         for license_file in ["COPYRIGHT", "COPYING", "CREDITS"]:
             copy(self, license_file, self.source_folder, os.path.join(self.package_folder, "licenses"))
-        if is_msvc(self):
-            cmake = CMake(self)
-            cmake.install()
-        else:
-            with chdir(self, self.source_folder):
-                autotools = Autotools(self)
-                autotools.install()
-            rm(self, "*.la", self.package_folder, recursive=True)
-            fix_apple_shared_install_name(self)
+
+        cmake = CMake(self)
+        cmake.install()
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
 
     def package_info(self):
-        if is_msvc(self):
-            self.cpp_info.libs = ["libid3tag"]
-            self.cpp_info.defines.append("ID3TAG_EXPORT=" + ("__declspec(dllimport)" if self.options.shared else ""))
-        else:
-            self.cpp_info.libs = ["id3tag"]
+        self.cpp_info.libs = ["id3tag"]
```

Comment:
```suggestion
        self.cpp_info.set_property("cmake_file_name", "id3tag")
        self.cpp_info.set_property("cmake_target_name", "id3tag::id3tag")
        self.cpp_info.set_property("pkg_config_name", "id3tag")
        self.cpp_info.libs = ["id3tag"]
```
I would recommend including this info even if it matches the defaults. It makes it clear that these are the actual official target names and not just implicit values generated by Conan.

Reply:
But the actual package name is ``libid3tag``, so these would be necessary if the upstream are defining the targets as ``id3tag``

Reply:
Oh, I missed that part, since I copied these from my older PR.
Either way, these are valid:
- https://codeberg.org/tenacityteam/libid3tag/src/branch/main/packaging
- https://packages.debian.org/trixie/amd64/libid3tag0-dev/filelist

Please do double check these details when reviewing, as it's going to be a breaking change for the consumers or will at least needs a "backwards compatible" fix when corrected later.